sql: TestExperimentalRelocateNonVoters failed

sql.TestExperimentalRelocateNonVoters [failed](https://tanzanite.cluster.engflow.com/invocation/19b917c6-82af-48d7-8d27-359d1af53e0b) on master @ [e747f6e6857a19d6048cb184b0f55c52cb8a6390](https://github.com/cockroachdb/cockroach/commits/e747f6e6857a19d6048cb184b0f55c52cb8a6390):


```
=== RUN   TestExperimentalRelocateNonVoters
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestExperimentalRelocateNonVoters2950769689
    test_log_scope.go:81: use -show-logs to present logs inline
    multitenant_admin_function_test.go:1002: -- test log scope end --
test logs left over in: outputs.zip/logTestExperimentalRelocateNonVoters2950769689
--- FAIL: TestExperimentalRelocateNonVoters (363.23s)
=== RUN   TestExperimentalRelocateNonVoters/ALTER_TABLE_x_EXPERIMENTAL_RELOCATE_NONVOTERS
    multitenant_admin_function_test.go:380: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:380
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            				github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:42
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:377
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:991
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:346
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:966
        	Error:      	Received unexpected error:
        	            	pq: while carrying out changes [{ADD_NON_VOTER n2,s2} {REMOVE_NON_VOTER n5,s5}]: change replicas of r68 failed: descriptor changed: [expected] r68:/Tenant/1{0-1} [(n1,s1):1, (n4,s4):2, (n3,s3):3, (n5,s5):4NON_VOTER, next=5, gen=7, sticky=1702118645.731465558,0] != [actual] r68:/Tenant/10{-/Table/104} [(n1,s1):1, (n4,s4):2, (n3,s3):3, (n5,s5):4NON_VOTER, next=5, gen=8, sticky=1702118645.731465558,0]
        	Test:       	TestExperimentalRelocateNonVoters/ALTER_TABLE_x_EXPERIMENTAL_RELOCATE_NONVOTERS
        	Messages:   	tenant=secondary query=``ALTER TABLE t EXPERIMENTAL_RELOCATE NONVOTERS VALUES (ARRAY[2], 1);`` leaseholder=4 replicas=[1 3 4 5] voting_replicas=[1 3 4] non_voting_replicas=[5] fromReplica=5 toReplica=2
    --- FAIL: TestExperimentalRelocateNonVoters/ALTER_TABLE_x_EXPERIMENTAL_RELOCATE_NONVOTERS (363.22s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=10</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExperimentalRelocateNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34311

sql.TestExperimentalRelocateNonVoters [failed](https://tanzanite.cluster.engflow.com/invocation/3e738b04-e5e2-484b-a3cd-c5a7464528d9) on master @ [e747f6e6857a19d6048cb184b0f55c52cb8a6390](https://github.com/cockroachdb/cockroach/commits/e747f6e6857a19d6048cb184b0f55c52cb8a6390):


```
=== RUN   TestExperimentalRelocateNonVoters
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestExperimentalRelocateNonVoters1618645633
    test_log_scope.go:81: use -show-logs to present logs inline
    multitenant_admin_function_test.go:1002: -- test log scope end --
test logs left over in: outputs.zip/logTestExperimentalRelocateNonVoters1618645633
--- FAIL: TestExperimentalRelocateNonVoters (1154.31s)
=== RUN   TestExperimentalRelocateNonVoters/ALTER_TABLE_x_EXPERIMENTAL_RELOCATE_NONVOTERS
    multitenant_admin_function_test.go:380: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:380
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            				github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60
        	            				github.com/cockroachdb/cockroach/pkg/testutils/soon.go:42
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:377
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:991
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:344
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/multitenant_admin_function_test.go:966
        	Error:      	Received unexpected error:
        	            	pq: while carrying out changes [{ADD_NON_VOTER n4,s4} {REMOVE_NON_VOTER n2,s2}]: change replicas of r65 failed: descriptor changed: [expected] r65:/T{able/63-enant/10} [(n1,s1):1, (n2,s2):6NON_VOTER, (n3,s3):5, (n5,s5):4, (n4,s4):7NON_VOTER, next=8, gen=27] != [actual] r65:/Table/{63-105} [(n1,s1):1, (n2,s2):6NON_VOTER, (n3,s3):5, (n5,s5):4, (n4,s4):7NON_VOTER, next=8, gen=28]
        	Test:       	TestExperimentalRelocateNonVoters/ALTER_TABLE_x_EXPERIMENTAL_RELOCATE_NONVOTERS
        	Messages:   	tenant=system query=``ALTER TABLE t EXPERIMENTAL_RELOCATE NONVOTERS VALUES (ARRAY[4], 1);`` leaseholder=1 replicas=[1 2 3 5] voting_replicas=[1 3 5] non_voting_replicas=[2] fromReplica=2 toReplica=4
    --- FAIL: TestExperimentalRelocateNonVoters/ALTER_TABLE_x_EXPERIMENTAL_RELOCATE_NONVOTERS (1154.30s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=10</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExperimentalRelocateNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/multitenant_admin_function_test.go
+++ b/pkg/sql/multitenant_admin_function_test.go
@@ -937,6 +937,7 @@ func TestRelocateNonVoters(t *testing.T) {
 func TestExperimentalRelocateNonVoters(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+	skip.UnderDuress(t, "test flakes in slow builds; see #108081")
 
 	testCases := []testCase{
 		{

